### PR TITLE
fix: create command adaptions

### PR DESCRIPTION
**Pull request by dennismeister93**
This PR adds adaptions to the create command

### CODE REVIEW

File: src/commands/create/index.ts [hunk 1]
```diff
@@ -117,27 +126,22 @@ export default class Create extends Command {
...
+            flags.example = flags.name = exampleResponse.exampleUse;
         } else {
-            interactiveResponses = await inquirer.prompt([Create.prompts.interface]);
-        }
-        flags.example = interactiveResponses.exampleUse;
-        flags.interface = interactiveResponses.interface;
-
-        if (flags.interface && flags.interface.length > 0) {
-            await this._handleAdditionalInterfaceArgs(flags.interface);
+            const interactiveNoExampleResponses: any = await inquirer.prompt([Create.prompts.name, Create.prompts.interface]);
```
Comment (doosuu) [inline src/commands/create/index.ts:139]: Better than calling this "noExamples" (so an inversion of the previous block) its easier to read and understand when you call it what it actually is: Responses for a skeleton or empty app
```suggestion
            const interactiveSkeletonAppResponses: any = await inquirer.prompt([Create.prompts.name, Create.prompts.interface]);
```

File: src/commands/create/index.ts [hunk 2]
```diff
@@ -117,27 +126,22 @@ export default class Create extends Command {
...
+            flags.name = interactiveNoExampleResponses.name;
+            flags.interface = interactiveNoExampleResponses.interface;
+            if (flags.interface && flags.interface.length > 0) {
```
Comment (doosuu) [inline src/commands/create/index.ts:142]: If possible, keep the empty line for a little visual separation.

File: src/commands/create/index.ts [hunk 3]
```diff
@@ -223,6 +211,10 @@ export default class Create extends Command {
         const { flags } = await this.parse(Create);
         this.log(`Creating a new Velocitas project ...`);
 
+        if (flags.name && flags.example) {
+            throw new Error("Flags 'name' and 'example' cannot be used in parallel");
```
Comment (doosuu) [inline src/commands/create/index.ts:215]: Small suggestion:
```suggestion
            throw new Error("Flags 'name' and 'example' are mutually exclusive!");
```

File: src/commands/create/index.ts [hunk 4]
```diff
@@ -236,7 +228,9 @@ export default class Create extends Command {
         }
 
         if (!flags.example) {
-            this._setDefaultAppManifestInterfaceConfig(flags.interface!);
+            if (flags.interface) {
```
Comment (doosuu) [inline src/commands/create/index.ts:231]: Join these `if`s into one statement to decrease nesting.

File: src/commands/create/index.ts [hunk 5]
```diff
@@ -185,28 +189,12 @@ export default class Create extends Command {
         });
     }
 
-    private async _setDefaultAppManifestInterfaceConfig(interfaces: string[]) {
-        if (this.appManifestInterfaces.interfaces.length > 0) {
+    private async _promptMissingInterfaceConfig(interfaces: string[]): Promise<void> {
```
Comment (doosuu) [inline src/commands/create/index.ts:192]: The argument name should convey what it really is. `interfaces` is very generic ;-)
```suggestion
    private async _promptMissingInterfaceConfig(inputInterfaces: string[]): Promise<void> {
```

Comment (doosuu) [inline src/commands/create/index.ts:192]: The actual code in this function is not really much. Can it maybe inlined somewhere?

File: test/commands/create/create.test.ts [hunk 6]
```diff
@@ -58,7 +58,9 @@ describe('create', () => {
         .stdout()
         .stub(gitModule, 'simpleGit', sinon.stub().returns(simpleGitInstanceMock()))
         .stub(exec, 'runExecSpec', () => {})
-        .stub(exec, 'awaitSpawn', () => {})
+        .stub(exec, 'awaitSpawn', () => {
+            return { exitCode: 0 };
+        })
         .command(['create', '-n', TEST_APP_NAME, '-l', 'test'])
         .it('creates project with provided flags and generates .velocitas.json and AppManifest', (ctx) => {
```
Comment (doosuu) [inline test/commands/create/create.test.ts:65]:
```suggestion
        .it('creates a project with provided flags and generates .velocitas.json and AppManifest', (ctx) => {
```

File: testbench/test-create/vehicle-app-template/package-index.json [hunk 7]
```diff
@@ -9,7 +9,7 @@
                 "args": [
                     {
                         "id": "src",
-                        "description": "URI or path to VSS json (Leave empty for default: v3.0)",
+                        "description": "URI or path to VSS json",
```
Comment (doosuu) [inline testbench/test-create/vehicle-app-template/package-index.json:12]: Why this change? Looked good in the prompt imo.

**Review by doosuu — CHANGES_REQUESTED** [review]

File: testbench/test-create/vehicle-app-template/package-index.json (hunk 7 above)
Comment (dennismeister93) [reply testbench/test-create/vehicle-app-template/package-index.json:12]: because due to the default value the prompt looks like this:
`Config 'src' for interface 'vehicle-signal-interface': URI or path to VSS json (https://github.com/COVESA/vehicle_signal_specification/releases/download/v3.0/vss_rel_3.0.json)`

Comment (doosuu) [reply testbench/test-create/vehicle-app-template/package-index.json:12]: OK👍

File: src/modules/project-config.ts [hunk 8]
```diff
@@ -35,7 +36,7 @@ export class ProjectConfig implements ProjectConfigOptions {
 
     // project-wide variable configuration
     variables: Map<string, any> = new Map<string, any>();
-    cliVersion: string | undefined;
+    cliVersion: string = 'v0.6.2';
```
Comment (doosuu) [inline src/modules/project-config.ts:39]: Is it possible to read this dynamically?

Comment (dennismeister93) [reply src/modules/project-config.ts:39]: i think it is only possible to use `this.config.version`in a Command class

**Review by doosuu — APPROVED** [review]
lgtm
